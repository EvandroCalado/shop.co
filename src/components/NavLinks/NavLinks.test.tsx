import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { NavLinks } from '.';

describe('<NavLinks />', () => {
  it('should render with default values', () => {
    render(<NavLinks />);

    expect(screen.getByRole('link', { name: 'shop' })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'shop' })).toHaveAttribute(
      'href',
      '/shop',
    );

    expect(screen.getByRole('link', { name: 'on sale' })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'on sale' })).toHaveAttribute(
      'href',
      '/on-sale',
    );

    expect(
      screen.getByRole('link', { name: 'new arrivals' }),
    ).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'new arrivals' })).toHaveAttribute(
      'href',
      '/new-arrivals',
    );

    expect(screen.getByRole('link', { name: 'brands' })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'brands' })).toHaveAttribute(
      'href',
      '/brands',
    );
  });

  it('should render a snapshot', () => {
    const { container } = render(<NavLinks />);

    expect(container).toMatchSnapshot();
  });
});
