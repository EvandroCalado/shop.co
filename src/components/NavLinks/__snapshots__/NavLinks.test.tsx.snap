// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<NavLinks /> > should render a snapshot 1`] = `
<div>
  <nav
    class="hidden items-center gap-2 capitalize md:flex"
  >
    <a
      class="cursor-pointer duration-150 hover:text-zinc-600"
      href="/shop"
    >
      shop
    </a>
    <a
      class="cursor-pointer duration-150 hover:text-zinc-600"
      href="/on-sale"
    >
      on sale
    </a>
    <a
      class="cursor-pointer duration-150 hover:text-zinc-600"
      href="/new-arrivals"
    >
      new arrivals
    </a>
    <a
      class="cursor-pointer duration-150 hover:text-zinc-600"
      href="/brands"
    >
      brands
    </a>
  </nav>
</div>
`;
